### ISSUE TITLE
[ili2fgdb] kryptische Benennung der Geometrie-Klassen

### ISSUE THREAD

**Issue by mizehnder** [claeis/ili2db#430]
Schemaimport von http://models.geo.admin.ch/ARE/Nutzungsplanung_V1_2.ili führt zu nicht nachvollziehbarer Benennung der Geometrie-Klassen:
![grafik](https://user-images.githubusercontent.com/82505753/139874053-658274bb-c34c-4d8c-9fdf-af9d93f287ab.png)
Folgende Parameter wurden genutzt:
--schemaimport --defaultSrsCode 2056 --fgdbXyResolution 0.00005 --fgdbXyTolerance 0.0004 --createEnumTabs --createBasketCol
ili2fgdb-Version: 4.5.0

In der Version 1.1 http://models.geo.admin.ch/ARE/replaced/Nutzungsplanung_V1_1.ili wurden die Geometrie-Klassen sinnvoll, gemäss Klassennamen, benannt:
![grafik](https://user-images.githubusercontent.com/82505753/139874641-be46755f-fa40-4962-b29d-3298eaa296e9.png)

Soweit ich sehe, wurden die Geometrie-Strukturen von 1.1 auf 1.2 nicht angepasst.
Wieso werden die Klassennamen nun anders abgebildet?

**Comment by edigonzales** [comment]
Es liegt einerseits an der Beziehung `ASSOCIATION Geometrie_Dokument`, die neu ist und andererseits, dass FGDB (und auch GPKG) nur eine Geometriespalte pro Tabelle erlaubt. So greift wohl die "smarte" Abbildungsregel nicht gleich gut und es führt zu einer für den Benutzer undankbaren Normalisierung.

Gegentest: Wenn ich die Assoziation entferne im 1.2er-Modell funktionierts wie anhin. Wenn ich die Daten nach Postgresql mit der Assoziation importiere, funktionierts wie anhin. Wenn ich beim Postgresql-Import `--oneGeomPerTable` verwende, erhalte ich die gleichen Tabellen wie bei FGDB.

**Comment by mizehnder** [comment]
Vielen Dank für die Tests Stefan! Nun kann ich nachvollziehen woran es liegt.

@claeis : Könnte man dann die Abbildungsregel entsprechend anpassen, damit auch in diesem Fall die Klassennamen sinnvoll benannt werden oder muss man aufgrund der Modellstruktur damit leben?

**Comment by claeis** [comment]
Im Prinzip ist es möglich, die automatisch generierten Namen zu übersteuern. Indem man die manuell definierten Namen in die Namensabbildungstabellen (t_ili2db_classname und t_ili2db_attrname)  einfügt (vor dem --schemaimport!).

**Comment by mizehnder** [comment]
Als Workaround gut zu Wissen, klingt aber aufwändig.
Kann denn die Abbildungsregel nicht entsprechend angepasst werden, damit auch in diesem Fall die Klassennamen übernommen werden?

**Comment by claeis** [comment]
Wie wären denn bessere Namen?

**Comment by edigonzales** [comment]
@mizehnder Geht es dir nur um die Namen oder dass die Tabellen in FGDB wie in der alten ili-Version abgebildet werden?

**Comment by mizehnder** [comment]
@edigonzales stimmt, es geht mir nicht nur um die Namen. Auch die Struktur verstehe ich nicht.

Bei V1.1 sind in jeder Geometrie-Tabelle die Attribute aus der Abstrakten Klasse Geometrie enthalten wie auch der Schlüssel zur Tabelle typ:
![grafik](https://user-images.githubusercontent.com/82505753/140711500-62fbc427-33d8-4167-94e3-c74e9eba0f5f.png)

Bei V1.2 sind diese Attribute nur in der Geometrie-Tabelle "geometrie" vorhanden:
![grafik](https://user-images.githubusercontent.com/82505753/140712012-ab8b6740-dc65-4660-97be-845a853035cf.png)

Bei den drei anderen Geometrie-Tabellen fehlen diese Attribute:
![grafik](https://user-images.githubusercontent.com/82505753/140712088-910cb78a-b907-4562-ac8e-fc106b3a2e15.png)

Wo finde ich diese Angaben? Wie kann ich die Beziehung zur Tabelle typ herstellen?

**Comment by edigonzales** [comment]
Hilfreich sind ER-Diagramme z.B. in dbeaver (wohl nicht für FGDB aber für GPKG). Dazu muss man `--createFk` verwendet werden. Gibt es sowas nicht in ArcWhatever? Und die Tabelle "t_ili2db_classname". Dort siehst du welche INTERLIS-Klasse zu welcher Tabelle gemappt wurde. 
Zwei Screenshots (Achtung: habe noch --nameByTopic verwendet).

![Screenshot 2021-11-08 at 10 47 38](https://user-images.githubusercontent.com/2767800/140720498-4483a8da-57e0-4d00-ac95-cbff594ab11b.png)

![Screenshot 2021-11-08 at 10 47 35](https://user-images.githubusercontent.com/2767800/140720527-0317159e-e84d-4cd3-82b5-e5dc1a5adf7e.png)

**Comment by mizehnder** [comment]
Lösung:
Mit --smart2Inheritance erfolgt die Tabellenabbildung gleich wie bei der Modellversion V1.1 und die Geometrie-Tabellen werden gemäss den Klassennamen benannt:
![grafik](https://user-images.githubusercontent.com/82505753/141285919-aaa9f034-14af-46f8-9c41-ed9eb796ee54.png)

Aktuell lässts sich zwar der Katalog nicht direkt importieren, das ist aber ein anderes Problem.
